### PR TITLE
Use PoI for channel transactions

### PR DESCRIPTION
**Pull request by velzevur**
[Closing txs provide proof of inclusion](https://www.pivotaltracker.com/story/show/157838691)
Channels keep state in a state tree structure, that is not persisted in DB. On-chain txs include its root hash and channel round so are part of the channel state on-chain. This allowes to optimize the signing process - since the create/withdraw/deposit transaction includes in itself the round and the state hash of this round, we don't need to sign the off-chain state as well but rather use this snapshot.
- [x] Rebase on top of master.
- [x] Write tests for closing after create/withdraw/deposit is included on-chain

### CODE REVIEW

**Comment by velzevur** [comment]
There had been a bug of create tx-s not being able to be used as a payload. It is resolved now.

**Comment by velzevur** [comment]
There is a race. From the [log](https://16527-99802036-gh.circle-artifacts.com/1/home/builder/epoch/_build/test/logs/ct_run.epoch_ct%40localhost.2018-06-13_14.59.44/lib.aechannel.aesc_fsm_SUITE.logs/run.2018-06-13_14.59.44/log_private/aesc_fsm_SUITE/dev1/log/epoch.log)
```
.0>@aesc_fsm_min_depth_watcher:determine_depth:104 tx at height 35
2018-06-13 15:01:08.458 [debug] <0.6583.0>@aesc_fsm_min_depth_watcher:determine_depth:104 tx at height 35
2018-06-13 15:01:08.458 [debug] <0.6590.0>@aesc_fsm_min_depth_watcher:determine_depth:104 tx at height 35
2018-06-13 15:01:08.458 [debug] <0.6589.0>@aesc_fsm_min_depth_watcher:determine_depth:104 tx at height 35
2018-06-13 15:01:08.458 [debug] <0.6575.0>@aesc_fsm_min_depth_watcher:determine_depth:104 tx at height 35
2018-06-13 15:01:08.458 [debug] <0.6570.0>@aesc_fsm_min_depth_watcher:determine_depth:104 tx at height 35
2018-06-13 15:01:08.458 [debug] <0.6578.0>@aesc_fsm_min_depth_watcher:determine_depth:104 tx at height 35
2018-06-13 15:01:08.458 [debug] <0.6577.0>@aesc_fsm_min_depth_watcher:determine_depth:104 tx at height 35
2018-06-13 15:01:08.458 [debug] <0.6582.0>@aesc_fsm_min_depth_watcher:determine_depth:104 tx at height 35
2018-06-13 15:01:08.458 [debug] <0.6568.0>@aesc_fsm_min_depth_watcher:determine_depth:104 tx at height 35
2018-06-13 15:01:08.458 [debug] <0.6576.0>@aesc_fsm_min_depth_watcher:determine_depth:104 tx at height 35
2018-06-13 15:01:08.458 [debug] <0.6579.0>@aesc_fsm_min_depth_watcher:determine_depth:104 tx at height 35
2018-06-13 15:01:08.458 [debug] <0.6574.0>@aesc_fsm_min_depth_watcher:determine_depth:104 tx at height 35
2018-06-13 15:01:08.458 [debug] <0.6573.0>@aesc_fsm_min_depth_watcher:determine_depth:104 tx at height 35
2018-06-13 15:01:08.458 [debug] <0.6587.0>@aesc_fsm_min_depth_watcher:determine_depth:104 tx at height 35
2018-06-13 15:01:08.458 [debug] <0.6567.0>@aesc_fsm_min_depth_watcher:determine_depth:104 tx at height 35
2018-06-13 15:01:08.458 [debug] <0.6572.0>@aesc_fsm_min_depth_watcher:determine_depth:104 tx at height 35
2018-06-13 15:01:08.458 [debug] <0.6581.0>@aesc_fsm_min_depth_watcher:determine_depth:104 tx at height 35
2018-06-13 15:01:08.458 [debug] <0.6584.0>@aesc_fsm_min_depth_watcher:determine_depth:104 tx at height 35
2018-06-13 15:01:08.458 [debug] <0.6580.0>@aesc_fsm_min_depth_watcher:determine_depth_:116 top height = 37
2018-06-13 15:01:08.458 [debug] <0.6580.0>@aesc_fsm_min_depth_watcher:check_status:79 min_depth not yet achieved
2018-06-13 15:01:08.458 [debug] <0.6569.0>@aesc_fsm_min_depth_watcher:determine_depth_:116 top height = 37
2018-06-13 15:01:08.458 [debug] <0.6569.0>@aesc_fsm_min_depth_watcher:check_status:79 min_depth not yet achieved
2018-06-13 15:01:08.459 [debug] <0.6590.0>@aesc_fsm_min_depth_watcher:determine_depth_:116 top height = 37
2018-06-13 15:01:08.459 [debug] <0.6590.0>@aesc_fsm_min_depth_watcher:check_status:79 min_depth not yet achieved
2018-06-13 15:01:08.459 [debug] <0.6583.0>@aesc_fsm_min_depth_watcher:determine_depth_:116 top height = 37
2018-06-13 15:01:08.459 [debug] <0.6583.0>@aesc_fsm_min_depth_watcher:check_status:79 min_depth not yet achieved
2018-06-13 15:01:08.459 [debug] <0.6575.0>@aesc_fsm_min_depth_watcher:determine_depth_:116 top height = 37
2018-06-13 15:01:08.459 [debug] <0.6589.0>@aesc_fsm_min_depth_watcher:determine_depth_:116 top height = 37
2018-06-13 15:01:08.459 [debug] <0.6575.0>@aesc_fsm_min_depth_watcher:check_status:79 min_depth not yet achieved
2018-06-13 15:01:08.459 [debug] <0.6589.0>@aesc_fsm_min_depth_watcher:check_status:79 min_depth not yet achieved
2018-06-13 15:01:08.459 [debug] <0.6578.0>@aesc_fsm_min_depth_watcher:determine_depth_:116 top height = 37
2018-06-13 15:01:08.459 [debug] <0.6578.0>@aesc_fsm_min_depth_watcher:check_status:79 min_depth not yet achieved
2018-06-13 15:01:08.459 [debug] <0.6582.0>@aesc_fsm_min_depth_watcher:determine_depth_:116 top height = 37
2018-06-13 15:01:08.459 [debug] <0.6577.0>@aesc_fsm_min_depth_watcher:determine_depth_:116 top height = 37
2018-06-13 15:01:08.459 [debug] <0.6582.0>@aesc_fsm_min_depth_watcher:check_status:79 min_depth not yet achieved
2018-06-13 15:01:08.459 [debug] <0.6577.0>@aesc_fsm_min_depth_watcher:check_status:79 min_depth not yet achieved
2018-06-13 15:01:08.459 [debug] <0.6568.0>@aesc_fsm_min_depth_watcher:determine_depth_:116 top height = 37
2018-06-13 15:01:08.459 [debug] <0.6579.0>@aesc_fsm_min_depth_watcher:determine_depth_:116 top height = 37
2018-06-13 15:01:08.459 [debug] <0.6568.0>@aesc_fsm_min_depth_watcher:check_status:79 min_depth not yet achieved
2018-06-13 15:01:08.459 [debug] <0.6579.0>@aesc_fsm_min_depth_watcher:check_status:79 min_depth not yet achieved
2018-06-13 15:01:08.459 [debug] <0.6574.0>@aesc_fsm_min_depth_watcher:determine_depth_:116 top height = 37
2018-06-13 15:01:08.459 [debug] <0.6574.0>@aesc_fsm_min_depth_watcher:check_status:79 min_depth not yet achieved
2018-06-13 15:01:08.459 [debug] <0.6572.0>@aesc_fsm_min_depth_watcher:determine_depth_:116 top height = 37
2018-06-13 15:01:08.459 [debug] <0.6587.0>@aesc_fsm_min_depth_watcher:determine_depth_:116 top height = 37
2018-06-13 15:01:08.459 [debug] <0.6572.0>@aesc_fsm_min_depth_watcher:check_status:79 min_depth not yet achieved
2018-06-13 15:01:08.459 [debug] <0.6573.0>@aesc_fsm_min_depth_watcher:determine_depth_:116 top height = 37
2018-06-13 15:01:08.459 [debug] <0.6587.0>@aesc_fsm_min_depth_watcher:check_status:79 min_depth not yet achieved
2018-06-13 15:01:08.459 [debug] <0.6573.0>@aesc_fsm_min_depth_watcher:check_status:79 min_depth not yet achieved
2018-06-13 15:01:08.459 [debug] <0.6584.0>@aesc_fsm_min_depth_watcher:determine_depth_:116 top height = 37
2018-06-13 15:01:08.459 [debug] <0.6570.0>@aesc_fsm_min_depth_watcher:determine_depth_:116 top height = 37
2018-06-13 15:01:08.459 [debug] <0.6584.0>@aesc_fsm_min_depth_watcher:check_status:79 min_depth not yet achieved
2018-06-13 15:01:08.459 [debug] <0.6576.0>@aesc_fsm_min_depth_watcher:determine_depth_:116 top height = 37
2018-06-13 15:01:08.459 [debug] <0.6570.0>@aesc_fsm_min_depth_watcher:check_status:79 min_depth not yet achieved
2018-06-13 15:01:08.459 [debug] <0.6567.0>@aesc_fsm_min_depth_watcher:determine_depth_:116 top height = 37
2018-06-13 15:01:08.459 [debug] <0.6576.0>@aesc_fsm_min_depth_watcher:check_status:79 min_depth not yet achieved
2018-06-13 15:01:08.459 [debug] <0.6567.0>@aesc_fsm_min_depth_watcher:check_status:79 min_depth not yet achieved
2018-06-13 15:01:08.459 [debug] <0.6581.0>@aesc_fsm_min_depth_watcher:determine_depth_:116 top height = 37
2018-06-13 15:01:08.459 [debug] <0.6581.0>@aesc_fsm_min_depth_watcher:check_status:79 min_depth not yet achieved
2018-06-13 15:01:08.461 [debug] <0.272.0>@aec_block_generator:handle_cast:74 New candidate generated```
The minimal depth is 3 and it looks like it is not reached until the [wild guess timeout](https://github.com/aeternity/epoch/blob/PT-157838691_closing_txs_poi/apps/aechannel/test/aesc_fsm_SUITE.erl#L247) is triggered

**Comment by velzevur** [comment]
Bumping the time interval for response from the FSM seems to fix it.

File: apps/aechannel/src/aesc_settle_tx.erl [hunk 1]
```diff
@@ -1,5 +1,3 @@
-%%%=============================================================================
-%%% @copyright 2018, Aeternity Anstalt
```
Comment (gorillainduction) [inline apps/aechannel/src/aesc_settle_tx.erl]: Probably a mistake?

File: apps/aehttp/test/aehttp_integration_SUITE.erl [hunk 2]
```diff
@@ -1162,14 +1169,17 @@ state_channels_deposit(ChannelId, MinerPubkey) ->
 
 state_channels_withdrawal(ChannelId, MinerPubkey) ->
     MinerAddress = aec_base58c:encode(account_pubkey, MinerPubkey),
-    MinerAddress = aec_base58c:encode(account_pubkey, MinerPubkey),
+    StateHash = <<123456>>,
```
Comment (gorillainduction) [inline apps/aehttp/test/aehttp_integration_SUITE.erl]: This shows that the state hashes aren't properly checked. Please add a size check on the root hash in the transactions.

File: apps/aetx/test/aetx_sign_tests.erl [hunk 3]
```diff
@@ -79,5 +79,7 @@ make_signed_mutual_close() ->
                                           responder_amount  => 24,
                                           ttl               => 1000,
                                           fee               => 10,
+                                          state_hash        => <<123456>>,
```
Comment (gorillainduction) [inline apps/aetx/test/aetx_sign_tests.erl]: Size check again, please ;-)

**Pull request merged**